client/my-sites/marketplace/pages/marketplace-plugin-upload-status/index.tsx

```diff
@@ -54,29 +72,66 @@ const MarketplacePluginUpload = (): JSX.Element => {
 	}, [ pluginUploadProgress, setCurrentStep ] );
 
 	useEffect( () => {
-		if ( pluginUploadComplete && uploadedPlugin ) {
+		if ( ! isUploadFlow && ! initializeInstallFlow && wporgPlugin && selectedSite ) {
+			// const upgradablePlan = isBusiness( selectedSite.plan ) || isEnterprise( selectedSite.plan ) || isEcommerce( selectedSite.plan );
+
+			if ( selectedSite?.jetpack ) {
+				// initialize plugin installing
+				dispatch( installPlugin( siteId, wporgPlugin, false ) );
+			} else {
+				// initialize atomic flow
+				setMoveToAtomicFlow( true );
+				dispatch( initiateTransfer( siteId, null, productSlug ) );
+			}
+
+			setInitializeInstallFlow( true );
+			waitFor( 1 ).then( () => setCurrentStep( 1 ) );
+		}
+	}, [ isUploadFlow, siteId, wporgPlugin, productSlug ] );
+
+	useEffect( () => {
+		if (
+			moveToAtomicFlow &&
+			currentStep === 1 &&
+			transferStates.COMPLETE === automatedTransferStatus
+		) {
+			setCurrentStep( 2 );
+		}
+	}, [ moveToAtomicFlow, automatedTransferStatus ] );
+
+	useEffect( () => {
+		if (
+			installedPlugin &&
+			currentStep === 1 &&
+			( ! isUploadFlow || ( isUploadFlow && pluginUploadComplete ) )
+		) {
 			dispatch(
-				togglePluginActivation( siteId, {
-					slug: uploadedPlugin?.slug,
-					id: uploadedPlugin?.id,
+				activatePlugin( siteId, {
+					slug: installedPlugin?.slug,
+					id: installedPlugin?.id,
 				} )
 			);
 			setCurrentStep( 2 );
 		}
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [ pluginUploadComplete, uploadedPlugin, setCurrentStep ] );
+	}, [ pluginUploadComplete, installedPlugin, setCurrentStep ] );
 
 	useEffect( () => {
-		if ( pluginActive ) {
+		if (
+			( installedPlugin && pluginActive ) ||
+			( moveToAtomicFlow && transferStates.COMPLETE === automatedTransferStatus )
+		) {
 			waitFor( 1 ).then( () =>
-				page( `/marketplace/thank-you/${ uploadedPlugin?.slug }/${ selectedSiteSlug }` )
+				page.redirect(
+					`/marketplace/thank-you/${ installedPlugin?.slug || productSlug }/${ selectedSiteSlug }`
+				)
 			);
 		}
```

Comment:
Since we added the install flow here together with the upload flow, can we add a small comment to each `userEffect` to signal why they are used?